src/openrct2/actions/RideEntranceExitPlaceAction.cpp

```diff
@@ -97,6 +97,11 @@ GameActions::Result RideEntranceExitPlaceAction::Query() const
         return GameActions::Result(GameActions::Status::NotOwned, errorTitle, STR_LAND_NOT_OWNED_BY_PARK);
     }
 
+    if (!map_is_location_in_park(_loc - CoordsDirectionDelta[_direction]))
+    {
+        return GameActions::Result(GameActions::Status::NotOwned, errorTitle, STR_FACING_PARK_FENCE);
+    }
+
```

Comment:
Can't test right now. Does this account for tiles with partial ownership? If so, I think it'd make sense to check if the entrance/exit height is at least the minimum building height there. The problem then is that when there is no path, guests will fall outside the park range anyway.

Reply:
The map_is_location_owned function seems to take that into account:

![image](https://user-images.githubusercontent.com/68320206/182910122-e47c635a-586b-4771-991e-0dfca3d2a9e1.png)

Do you know of a park that has those specific building rights?